src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -65,7 +87,60 @@ public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceReques
 				}
 			}
 
-			return base.ShouldOverrideUrlLoading(view, request);
+			return false;
+		}
+
+		private void TryStartActivityFromUri(string? uriString, bool uriHasIntentScheme)
+		{
+			if (_webViewHandler is null || uriString is null)
+			{
+				return;
+			}
+
+			try
+			{
+				var intent = Intent.ParseUri(uriString, IntentUriType.Scheme);
+
+				if (intent is not null)
+				{
+					_webViewHandler.Context.StartActivity(intent);
+				}
+			}
+			catch (URISyntaxException)
+			{
+				// No-op. The URI was malformed.
+				return;
+			}
+			catch (ActivityNotFoundException)
+			{
+				if (!uriHasIntentScheme)
+				{
+					// No-op. This behavior is consistent with other platforms when a URI with an unknown scheme is clicked.
+					return;
+				}
+
+				// Chrome for Android allows specifying a fallback URL in case an intent cannot be resolved.
+				// Since the Android WebView cannot handle "intent"-schemed URIs, we add our own support for the fallback
+				// URL here.
+				// See: https://developer.chrome.com/docs/multidevice/android/intents/
+
+				const string FallbackUrlQueryParameterPrefix = "S.browser_fallback_url=";
+
+				var fallbackUrlQueryParameterPrefixLocation = uriString.LastIndexOf(FallbackUrlQueryParameterPrefix);
+
+				if (fallbackUrlQueryParameterPrefixLocation != -1)
+				{
+					var fallbackUrlQueryParameterValueStart = fallbackUrlQueryParameterPrefixLocation + FallbackUrlQueryParameterPrefix.Length;
+					var fallbackUrlQueryParameterValueEnd = uriString.IndexOf(';', fallbackUrlQueryParameterValueStart);
+
+					if (fallbackUrlQueryParameterValueEnd != -1)
+					{
+						var encodedFallbackUrlString = uriString[fallbackUrlQueryParameterValueStart..fallbackUrlQueryParameterValueEnd];
+						var fallbackUrlString = HttpUtility.UrlDecode(encodedFallbackUrlString);
+						TryStartActivityFromUri(fallbackUrlString, uriHasIntentScheme: false);
```

Comment:
Wouldn't we want to reroute the fallback URI to `TryOverrideHttpUrlLoading`?

Reply:
You could have fallback URIs that don't have HTTP(S) schemes. For example, if you're trying to create a link that opens the YouTube app, you might want to open the Google Play store instead using the `market` scheme (e.g. "market://details?id=com.google.android.youtube").

Reply:
Coming back to this, I wonder if there's a scenario where you want the fallback link to be another page in the app? Because if that's the case, we probably would want to at least check if the scheme is `http` in case we want the specialized navigation behavior.

Reply:
(And this all depends on whether we think supporting the chrome `S.browser_fallback_url` extra is worthwhile)